pkg/api/handlers/libpod/images.go

```diff
@@ -498,12 +503,95 @@ func CommitContainer(w http.ResponseWriter, r *http.Request) {
 	if len(query.Repo) > 0 {
 		destImage = fmt.Sprintf("%s:%s", query.Repo, tag)
 	}
-	commitImage, err := ctr.Commit(r.Context(), destImage, options)
+
+	if !query.Stream {
+		commitImage, err := ctr.Commit(r.Context(), destImage, options)
+		if err != nil && !strings.Contains(err.Error(), "is not running") {
+			utils.Error(w, http.StatusInternalServerError, fmt.Errorf("CommitFailure: %w", err))
+			return
+		}
+		utils.WriteResponse(w, http.StatusOK, entities.IDResponse{ID: commitImage.ID()})
+		return
+	}
+
+	// Channels all mux'ed in select{} below to follow API commit protocol
+	stdout := channel.NewWriter(make(chan []byte))
+	defer stdout.Close()
+	stderr := channel.NewWriter(make(chan []byte))
```

Comment:
IMO this should just be a `chan error` so we could - in theory - do error checks later on.